docs/cpan-dependency-confusion.md

```diff
@@ -0,0 +1,238 @@
+- per file `mirror ""` is not used for index by carton
+- `PERL_CARTON_MIRROR="https://pause.local/" carton install` - safe but error prone (mirror is not explicited in requirements file)
+
+#### cpanfile/carmel
+- carmel is "experimental"
+- per req `mirror =>` / `dist =>` is [not supported](https://github.com/miyagawa/Carmel/issues/72)
+- per file `mirror ""` feature is [not stable](https://github.com/miyagawa/Carmel/issues/55) and behaviour [not decided yet](https://github.com/miyagawa/Carmel/pull/92)
+
+### Safe
+- `cpanm --from` - but requires full sync DarkPAN or to differenciate private and public cpanfile(s)
+- `unset PERL_CPANM_OPT cpanm --mirror-only --mirror darkpan --mirror cpan` - beware order of mirrors
+- `unset PERL_CPANM_OPT cpanm --mirror-index file:///data/02packages --mirror darkpan --mirror cpan` - beware order of mirrors
+- `unset PERL_CPANM_OPT cpanm --cascade-search --mirror-index file:///data/02packages --mirror darkpan --mirror cpan` - beware order of mirrors
+- `cpm --no-default-resolvers --resolver darkpan --resolver cpan` 
+- carton/cpanfile `mirror =>` AND `dist =>` - pinned version and storage
+- carton/cpanfile `url =>` - pinned url
+- CPAN urllist (but with `pushy_https` disabled) - use index of first mirror, fallback for download
+- CPANPLUS `hosts` - use index of first mirror, fallback for download
+
+Carton pinning with `url =>`:
+```
+requires 'Private', '0',
+  url => 'https://darkpan.local/authors/id/U/US/USER/Private-0.01.tar.gz';
+```
+
+## Deployment and lockfiles
+Lockfiles mostly refers to *deployment* phase that is out of scope of this document.
+
+`carton --deployment` and  `carton --cached` are not vulnerable, because the use dumped/recreated local index with local artefacts.
+
+## CPAN Artifacts Repositories
+### JFrog Artifactory
+JFrog is known to be vulnerable (internally) to this dependency confusion attacks with other ecosystems.
+
+Since JFrog Artifactory does not support CPAN as first class citizen but only "proxy" CPAN (as in "HTTP proxy"), CPAN is not vulnerable via this artifact repository manager.
+
+### StratoPAN
+Not affected, not doing fallback on public CPAN.
+
+## Mitigations
+Beyond technical CPAN client/options recommendations exposed earlier, sharing some more "shift left" guards and explicit security advices.
+
+### PAUSE operational model 
+The way CPAN and PAUSE work, and the very "social" nature of this ecosystem makes it very difficult to attack CPAN via dependency confusions. 
+
+1. New authors don't get immediate account but are approved by an admin. With a priori approval like this, automatic creation of fake accounts is hence not possible.
+2. New uploads are not following an a priori control but publicly reviewed. Malware code can be spotted and claiming namespace with bad intention would be very likely reported rapidly. 
+
+In this context, it sounds difficult for an attacker/malware module to go unnoticed and exploit dependency confusion.
```

Comment:
We've had onslaughts of spammy modules uploaded to CPAN before, on a weekend (or holiday time like Easter/Chrismas?). It's been a long time ago, and I can't find the discussions in the archive of the modules@perl.org list. But it's worth picking peoples' brains about as there are procedures for handling that.

I remember this being before 2013, I see this happened in August 2018 https://www.nntp.perl.org/group/perl.modules/2018/08.html

Reply:
Added a paragraph to mention the actual existence of some "spam raids" over the PAUSE/CPAN history